docs/src/content/docs/guides/sidebar.mdx

```diff
@@ -0,0 +1,461 @@
+					link: '/guides/components/',
+					badge: 'New',
+				},
+			],
+		},
+	],
+});
+```
+
+The above configuration will generate the following sidebar:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'New', variant: 'default' },
+				},
+			],
+		},
+	]}
+/>
+
+The `badge` property includes a `text` property (the content to display in the badge) and optionally a `variant` property (to define the type of badge shown).
+
+`badge.text` - a string representing the content to display. (e.g. "New")
+
+`badge.variant` - to override the `default` styling setting, set to one of the following properties for custom styling: `note`, `tip`, `danger`, `caution` or `success`.
+
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'Guides',
+			items: [
+				// A link with a green "New" badge.
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'Experimental', variant: 'caution' },
+				},
+			],
+		},
+	],
+});
+```
+
+The above configuration will generate the following sidebar:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'Experimental', variant: 'caution' },
+				},
+			],
+		},
+	]}
+/>
+
+## Autogenerated groups
+
+Any group (or sub-group) can be automatically generated, and will be sorted alphabetically by filename. This is helpful when you do not want to manually enter each sidebar item in a group.
+
+To create an autogenerated group, set `autogenerate` to an object specifying the `directory`. For example, given the following file structure:
+
+<FileTree>
+
+- src/
+  - content/
+    - docs/
+      - guides/
+        - components.md
+        - i18n.md
+        - advanced/
+          - project-structure.md
+
+</FileTree>
+
+And the following configuration:
+
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'Guides',
+			// Autogenerate a group of links for the 'guides' directory.
+			autogenerate: { directory: 'guides' },
+		},
+	],
+});
+```
+
+The following sidebar will be generated:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{ label: 'Components', link: '/guides/components/' },
+				{ label: 'Internationalization (i18n)', link: '/guides/i18n/' },
+				{
+					label: 'advanced',
+					items: [
+						{ label: 'Project Structure', link: '/guides/project-structure/' },
+					],
+				},
+			],
+		},
+	]}
+/>
+
+### Customizing autogenerated links in frontmatter
+
+Use the [`sidebar` frontmatter field](/reference/frontmatter/#sidebar) to customize autogenerated links.
+
+This field allows you to set a [custom label](/reference/frontmatter/#label) or add a [badge](/reference/frontmatter/#badge) for a page link, to [hide](/reference/frontmatter/#hidden) a link from the sidebar or to define a [custom order](/reference/frontmatter/#order) for all page links.
```

Comment:
Nit: I try to avoid things like “this” that refer back to previous context as much as possible especially across paragraphs. Also suggesting “sort weighting” vs “order” because although the property is named `order` it doesn’t really set the _order_ necessarily 😅 

```suggestion
Sidebar frontmatter options allow you to set a [custom label](/reference/frontmatter/#label) or add a [badge](/reference/frontmatter/#badge) to a link, [hide](/reference/frontmatter/#hidden) a link from the sidebar, or define a [custom sort weighting](/reference/frontmatter/#order).
```